### ISSUE TITLE
Latency spikes during 100% write test (perf-regression)

### ISSUE THREAD

**Issue by roydahan** [scylladb/scylla#6378]
*Installation details*
Scylla version (or git commit hash): 4.0.rc3
Cluster size: 3
OS (RHEL/CentOS/Ubuntu/AWS AMI): AWS AMI

Version: 4.0.rc3
build date: 20200501
commit id: eee4c00e29

<img width="790" alt="Screen Shot 2020-05-05 at 23 42 11" src="https://user-images.githubusercontent.com/20959584/81114125-13b13380-8f2a-11ea-961b-452f3971af88.png">

Full dashboard screenshot:
https://snapshot.raintank.io/dashboard/snapshot/s16w9bbPLuvBr6mFK9MqlJq3acYCxZBu?orgId=2

```
< t:2020-05-05 15:12:34,108 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:12:32+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 3.
< t:2020-05-05 15:12:34,126 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:12:32+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 3.
< t:2020-05-05 15:12:49,127 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 020-05-05T15:12:35+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 7.
< t:2020-05-05 15:12:49,138 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:12:39+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 0.
< t:2020-05-05 15:12:49,161 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 020-05-05T15:12:35+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 7.
< t:2020-05-05 15:12:49,174 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:12:39+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 0.
< t:2020-05-05 15:13:34,185 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 020-05-05T15:13:22+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 13.
< t:2020-05-05 15:13:34,229 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 020-05-05T15:13:22+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 13.
< t:2020-05-05 15:13:46,553 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 020-05-05T15:13:31+00:00  perf-regression-latency-perf-v8-db-node-2e908268-2 !INFO    | scylla: Reactor stalled for 5 ms on shard 1.
< t:2020-05-05 15:13:46,584 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 020-05-05T15:13:31+00:00  perf-regression-latency-perf-v8-db-node-2e908268-2 !INFO    | scylla: Reactor stalled for 5 ms on shard 1.
< t:2020-05-05 15:14:12,622 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:13:59+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 0.
< t:2020-05-05 15:14:12,626 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:14:01+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 3.
< t:2020-05-05 15:14:12,647 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:13:59+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 0.
< t:2020-05-05 15:14:12,659 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:14:01+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 3.
< t:2020-05-05 15:14:27,661 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:14:19+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 11.
< t:2020-05-05 15:14:27,708 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:14:19+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 11.
< t:2020-05-05 15:15:31,651 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:15:21+00:00  perf-regression-latency-perf-v8-db-node-2e908268-2 !INFO    | scylla: Reactor stalled for 5 ms on shard 2.
< t:2020-05-05 15:15:31,675 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:15:21+00:00  perf-regression-latency-perf-v8-db-node-2e908268-2 !INFO    | scylla: Reactor stalled for 5 ms on shard 2.
< t:2020-05-05 15:26:28,027 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:26:16+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 11.
< t:2020-05-05 15:26:28,052 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:26:16+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 11.
< t:2020-05-05 15:28:58,121 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:28:51+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 3.
< t:2020-05-05 15:28:58,163 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:28:51+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 3.
< t:2020-05-05 15:29:13,163 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 020-05-05T15:28:58+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 3.
< t:2020-05-05 15:29:13,182 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 020-05-05T15:28:58+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 3.
< t:2020-05-05 15:31:02,272 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:30:49+00:00  perf-regression-latency-perf-v8-db-node-2e908268-2 !INFO    | scylla: Reactor stalled for 5 ms on shard 12.
< t:2020-05-05 15:31:02,305 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:30:49+00:00  perf-regression-latency-perf-v8-db-node-2e908268-2 !INFO    | scylla: Reactor stalled for 5 ms on shard 12.
< t:2020-05-05 15:31:04,819 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:30:57+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 7.
< t:2020-05-05 15:31:04,836 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:30:57+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 5 ms on shard 7.
< t:2020-05-05 15:44:43,651 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:44:32+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 10.
< t:2020-05-05 15:44:43,671 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:44:32+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 10.
< t:2020-05-05 15:48:43,768 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:48:36+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 6 ms on shard 1.
< t:2020-05-05 15:48:43,785 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:48:36+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 6 ms on shard 1.
< t:2020-05-05 15:56:13,964 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T15:56:02+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 8.
< t:2020-05-05 15:56:14,005 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T15:56:02+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 8.
< t:2020-05-05 16:01:59,201 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T16:01:53+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 11 ms on shard 8.
< t:2020-05-05 16:01:59,222 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T16:01:53+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 11 ms on shard 8.
< t:2020-05-05 16:03:48,359 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T16:03:37+00:00  perf-regression-latency-perf-v8-db-node-2e908268-2 !INFO    | scylla: Reactor stalled for 11 ms on shard 10.
< t:2020-05-05 16:03:48,379 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T16:03:37+00:00  perf-regression-latency-perf-v8-db-node-2e908268-2 !INFO    | scylla: Reactor stalled for 11 ms on shard 10.
< t:2020-05-05 16:04:44,321 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T16:04:34+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 4.
< t:2020-05-05 16:04:44,345 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T16:04:34+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 4.
< t:2020-05-05 16:05:20,897 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T16:05:18+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 12 ms on shard 2.
< t:2020-05-05 16:05:20,900 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T16:05:18+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 22 ms on shard 2.
< t:2020-05-05 16:05:20,916 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T16:05:18+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 12 ms on shard 2.
< t:2020-05-05 16:05:20,935 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T16:05:18+00:00  perf-regression-latency-perf-v8-db-node-2e908268-3 !INFO    | scylla: Reactor stalled for 22 ms on shard 2.
< t:2020-05-05 16:06:59,417 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T16:06:47+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 11 ms on shard 10.
< t:2020-05-05 16:06:59,421 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T16:06:47+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 21 ms on shard 10.
< t:2020-05-05 16:06:59,444 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T16:06:47+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 11 ms on shard 10.
< t:2020-05-05 16:06:59,462 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T16:06:47+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 21 ms on shard 10.
< t:2020-05-05 16:09:29,618 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 2020-05-05T16:09:28+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 12.
< t:2020-05-05 16:09:29,637 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 2020-05-05T16:09:28+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 12.
< t:2020-05-05 16:10:14,661 f:cluster.py      l:1371 c:sdcm.cluster         p:DEBUG > 020-05-05T16:10:01+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 9.
< t:2020-05-05 16:10:14,685 f:sct_events.py   l:780  c:sdcm.sct_events      p:INFO  > 020-05-05T16:10:01+00:00  perf-regression-latency-perf-v8-db-node-2e908268-1 !INFO    | scylla: Reactor stalled for 5 ms on shard 9.
```

Decoded backtrace:
```
[centos@perf-regression-latency-perf-v8-db-node-2e908268-1 ~]$ addr2line -Cpfie /usr/lib/debug/.build-id/0f/e095b80ee54c7f47a688aaac0c6bb835118ddc.debug

?? ??:0

?? ??:0
0x0000000002ed8bc2
0x0000000002e7e73b
0x0000000002e7ec62
0x0000000002e7ef97
0x00007f2a8ea53b1f
0x0000000001deb872
0x0000000000fa2009
0x0000000000fa6567
0x0000000002e36708
0x0000000002e370d0
0x0000000002e3ab6f
0x00000000011ed7cd
0x000000000114606f
0x0000000001295a2e
0x000000000132dcdf
0x0000000001345c81
0x0000000001346237
0x000000000313b4fc
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/include/fmt/format.h:1158
seastar::backtrace_buffer::append_backtrace() at /usr/include/fmt/format.h:1158
 (inlined by) print_with_backtrace at ./build/release/seastar/./seastar/src/core/reactor.cc:755
seastar::internal::cpu_stall_detector::generate_trace() at /usr/include/fmt/format.h:1158
seastar::internal::cpu_stall_detector::maybe_report() at /usr/include/fmt/format.h:1158
 (inlined by) seastar::internal::cpu_stall_detector::on_signal() at ./build/release/seastar/./seastar/src/core/reactor.cc:1062
 (inlined by) seastar::reactor::block_notifier(int) at ./build/release/seastar/./seastar/src/core/reactor.cc:1185
?? ??:0
boost::range_iterator<std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > const, void>::type boost::range_detail::range_begin<std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > const>(std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > const&) at /usr/include/c++/9/bits/stl_vector.h:1756
 (inlined by) boost::range_iterator<std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > const, void>::type boost::range_adl_barrier::begin<std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > >(std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > const&) at /usr/include/boost/range/begin.hpp:119
 (inlined by) boost::range_detail::reversed_range<std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > const>::reversed_range(std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > const&) at /usr/include/boost/range/adaptor/reversed.hpp:42
 (inlined by) boost::range_detail::reversed_range<std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > const> boost::range_detail::operator|<std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > >(std::vector<std::vector<unsigned long, std::allocator<unsigned long> >, std::allocator<std::vector<unsigned long, std::allocator<unsigned long> > > > const&, boost::range_detail::reverse_forwarder) at /usr/include/boost/range/adaptor/reversed.hpp:65
 (inlined by) utils::dynamic_bitset::find_first_set() const at ./utils/dynamic_bitset.cc:59
logalloc::segment_pool::reclaim_segments(unsigned long) at /usr/include/fmt/format.h:1158
logalloc::tracker::impl::reclaim(unsigned long) at /usr/include/fmt/format.h:1158
 (inlined by) logalloc::tracker::reclaim(unsigned long) at ./utils/logalloc.cc:418
 (inlined by) logalloc::tracker::reclaim(seastar::memory::reclaimer::request) at ./utils/logalloc.cc:1679
 (inlined by) operator() at ./utils/logalloc.cc:411
 (inlined by) _M_invoke at /usr/include/c++/9/bits/std_function.h:285
std::function<seastar::memory::reclaiming_result (seastar::memory::reclaimer::request)>::operator()(seastar::memory::reclaimer::request) const at /usr/include/c++/9/ostream:221
 (inlined by) seastar::memory::reclaimer::do_reclaim(unsigned long) at ./build/release/seastar/./seastar/include/seastar/core/memory.hh:129
 (inlined by) seastar::memory::cpu_pages::run_reclaimers(seastar::memory::reclaimer_scope, unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1034
seastar::memory::cpu_pages::maybe_reclaim() at /usr/include/c++/9/ostream:221
 (inlined by) seastar::memory::cpu_pages::maybe_reclaim() at ./build/release/seastar/./seastar/src/core/memory.cc:613
 (inlined by) seastar::memory::cpu_pages::allocate_large_and_trim(unsigned int) at ./build/release/seastar/./seastar/src/core/memory.cc:656
seastar::memory::cpu_pages::allocate_large(unsigned int) at /usr/include/c++/9/ostream:221
 (inlined by) seastar::memory::allocate_large(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1216
 (inlined by) seastar::memory::allocate(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1268
 (inlined by) malloc at ./build/release/seastar/./seastar/src/core/memory.cc:1504
utils::chunked_vector<unsigned int, 131072ul>::new_chunk(unsigned long) at /usr/include/fmt/format.h:1158
 (inlined by) utils::chunked_vector<unsigned int, 131072ul>::make_room(unsigned long) at ././utils/chunked_vector.hh:380
utils::chunked_vector<unsigned int, 131072ul>::reserve(unsigned long) at /usr/include/fmt/format.h:1158
 (inlined by) utils::chunked_vector<unsigned int, 131072ul>::do_reserve_for_push_back() at ././utils/chunked_vector.hh:396
 (inlined by) utils::chunked_vector<unsigned int, 131072ul>::reserve_for_push_back() at ././utils/chunked_vector.hh:61
 (inlined by) utils::chunked_vector<unsigned int, 131072ul>::push_back(unsigned int&&) at ././utils/chunked_vector.hh:122
 (inlined by) sstables::seal_summary(sstables::summary_ka&, std::optional<sstables::key>&&, std::optional<sstables::key>&&, sstables::index_sampling_state const&) at ./sstables/sstables.cc:1843
sstables::mc::writer::consume_end_of_stream() at ././sstables/types.hh:261
sstables::compaction::finish_new_sstable(std::optional<sstables::sstable_writer>&, seastar::lw_shared_ptr<sstables::sstable>&) at /usr/include/fmt/format.h:1158
 (inlined by) sstables::regular_compaction::stop_sstable_writer() at ./sstables/compaction.cc:780
 (inlined by) sstables::regular_compaction::finish_sstable_writer() at ./sstables/compaction.cc:787
sstables::compacting_sstable_writer::consume_end_of_stream() at ./sstables/compaction.cc:642
 (inlined by) auto compact_mutation_state<(emit_only_live_rows)0, (compact_for_sstables)1>::consume_end_of_stream<sstables::compacting_sstable_writer, noop_compacted_fragments_consumer>(sstables::compacting_sstable_writer&, noop_compacted_fragments_consumer&) at ././mutation_compactor.hh:424
 (inlined by) compact_mutation<(emit_only_live_rows)0, (compact_for_sstables)1, sstables::compacting_sstable_writer, noop_compacted_fragments_consumer>::consume_end_of_stream() at ././mutation_compactor.hh:539
 (inlined by) stable_flattened_mutations_consumer<compact_for_compaction<sstables::compacting_sstable_writer, noop_compacted_fragments_consumer> >::consume_end_of_stream() at ././mutation_reader.hh:350
 (inlined by) auto flat_mutation_reader::impl::consume_in_thread<stable_flattened_mutations_consumer<compact_for_compaction<sstables::compacting_sstable_writer, noop_compacted_fragments_consumer> >, flat_mutation_reader::filter>(stable_flattened_mutations_consumer<compact_for_compaction<sstables::compacting_sstable_writer, noop_compacted_fragments_consumer> >, flat_mutation_reader::filter, std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at ././flat_mutation_reader.hh:280
 (inlined by) auto flat_mutation_reader::consume_in_thread<stable_flattened_mutations_consumer<compact_for_compaction<sstables::compacting_sstable_writer, noop_compacted_fragments_consumer> >, flat_mutation_reader::filter>(stable_flattened_mutations_consumer<compact_for_compaction<sstables::compacting_sstable_writer, noop_compacted_fragments_consumer> >, flat_mutation_reader::filter, std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >) at ././flat_mutation_reader.hh:399
 (inlined by) seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}::operator()() at ./sstables/compaction.cc:1284
seastar::apply_helper<seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}, std::tuple<>&&, std::integer_sequence<unsigned long> >::apply({lambda()#1}&&, std::tuple<>) at ././seastar/include/seastar/core/apply.hh:36
 (inlined by) auto seastar::apply<seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}>(seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}&&, std::tuple<>&&) at ././seastar/include/seastar/core/apply.hh:44
 (inlined by) seastar::future<sstables::compaction_info> seastar::futurize<sstables::compaction_info>::apply<seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}>(seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}&&, std::tuple<>&&) at ././seastar/include/seastar/core/future.hh:1584
 (inlined by) seastar::futurize<std::result_of<std::decay<seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}>::type ()>::type>::type seastar::async<seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}>(seastar::thread_attributes, seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}&&, (std::decay<seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}>::type&&)...)::{lambda(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type> seastar::async<{lambda()#1}>(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type>::type, seastar::thread_attributes, std::decay<{lambda()#1}>::type&&)::work&)#1}::operator()(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type> seastar::async<{lambda()#1}>(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type>::type, seastar::thread_attributes, std::decay<{lambda()#1}>::type&&)::work)::{lambda()#1}::operator()() const at ././seastar/include/seastar/core/thread.hh:253
 (inlined by) seastar::noncopyable_function<void ()>::direct_vtable_for<seastar::futurize<std::result_of<std::decay<seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}>::type ()>::type>::type seastar::async<seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}>(seastar::thread_attributes, seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}&&, (std::decay<seastar::future<sstables::compaction_info> sstables::compaction::run<noop_compacted_fragments_consumer>(std::unique_ptr<sstables::compaction, std::default_delete<sstables::compaction> >, noop_compacted_fragments_consumer)::{lambda()#1}>::type&&)...)::{lambda(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type> seastar::async<{lambda()#1}>(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type>::type, seastar::thread_attributes, std::decay<{lambda()#1}>::type&&)::work&)#1}::operator()(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type> seastar::async<{lambda()#1}>(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type>::type, seastar::thread_attributes, std::decay<{lambda()#1}>::type&&)::work)::{lambda()#1}>::call(seastar::noncopyable_function<void ()> const*) at ././seastar/include/seastar/util/noncopyable_function.hh:101
seastar::noncopyable_function<void ()>::operator()() const at ./build/release/seastar/./seastar/include/seastar/util/noncopyable_function.hh:184
 (inlined by) seastar::thread_context::main() at ./build/release/seastar/./seastar/src/core/thread.cc:286

?? ??:0
```

**Comment by roydahan** [comment]
Probably the same as #6377 but @slivne wanted a separate issue for 4.0

**Comment by roydahan** [comment]
Live monitor in case it helps somehow: http://52.201.226.72:3000/d/xrD-mL6Wz/scylla-per-server-metrics-nemesis-master?orgId=1&from=1588691434396&to=1588695255801

**Comment by slivne** [comment]
I added stalls to the view - there is correlation with stall generation (which is good !!!) - yet some latency seems to be non related to stalls 

![stalls](https://user-images.githubusercontent.com/3465480/81115666-c08cb000-8f2c-11ea-9ccd-a1d288b3a73a.png)


Lets fix the stalls and then see what is left.

**Comment by slivne** [comment]
we fixed many stalls ove the last few month - closing this as we have dedicated testing for latency

**Comment by avikivity** [comment]
This was fixed by f5ffd5fc5fe1e70e398e4e173ef2af623b1cb5e6.
